ivy/func_wrapper.py

```diff
@@ -294,6 +295,43 @@ def _check_in_nested_sequence(sequence, value=None, _type=None):
             )
 
 
+# Invalid Backend Handling #
+# ---------------#
+
+
+def handle_backend_invalid(fn: Callable) -> Callable:
+    @functools.wraps(fn)
+    def _handle_backend_invalid(*args, **kwargs):
+        if args:
```

Comment:
I think a relatively general approach that also captures `ivy.Array` and `ivy.NativeArray` instances in `ivy.Container` arguments would be something along the lines of,
```
array_indices = ivy.nested_argwhere([args, kwargs], ivy.is_array)
array_vals = ivy.multi_index_nest([args, kwargs], array_indices)

def func(x):
    target_backend = ivy.utils.backend.handler._determine_backend_from_args(a)
    return target_backend is None and ivy.current_backend() != target_backend

if ivy.nested_any([array_vals], func):
    raise ivy.utils.exceptions.InvalidBackendException(
        "Operation not allowed. Array was instantiated with backend"
        f" {target_backend}. But current backend is"
        f" {ivy.current_backend_str()}. Please set dynamic=True"
        " for the array if you want to convert it to the target"
        " backend"
    )

return fn(*args, **kwargs)
```
Probably you're doing this already, but we'd skip all functions which don't accept `ivy.Array` in their arguments, let me know if the example you shared fails after making these changes, thanks @mosesdaudu001, let's see what @CatB1t thinks 😄

Reply:
Just a correction here as there's a couple of minor issues with the block I shared above, the code-block should instead look something like,
```
        array_indices = ivy.nested_argwhere([args, kwargs], lambda x: isinstance(x, ivy.Array))
        array_vals = ivy.multi_index_nest([args, kwargs], array_indices)

        def func(x):
            target_backend = ivy.utils.backend.handler._determine_backend_from_args(x)
            if target_backend is not None and ivy.current_backend() != target_backend:
                raise ivy.utils.exceptions.InvalidBackendException(
                    f"Operation not allowed. Array was instantiated with backend"
                    f" {target_backend.backend}. But current backend is"
                    f" {ivy.current_backend_str()}. Please set dynamic=True"
                    " for the array if you want to convert it to the target"
                    " backend"
                )
            return x

        ivy.nested_map(array_vals, func, include_derived=True)
        return fn(*args, **kwargs)
```